### PR TITLE
Update 'pipeline components' topic in the 'user guide'

### PR DESCRIPTION
**Pull request by ptitzler**
This PR updates the https://elyra.readthedocs.io/en/latest/user_guide/pipeline-components.html topic based on the changes delivered by https://github.com/elyra-ai/elyra/pull/2241
- introduce component catalogs and component catalog connectors
- link to the component catalog marketplace
- link to the BYO component catalog tutorial in the examples repository
- update the custom component management reference
- add reference for built-in component connectors

### What changes were proposed in this pull request?

See above.

Some of the HREFs to the examples repository will be broken until https://github.com/elyra-ai/examples/pull/78 was merged.

### How was this pull request tested?

Reviewed the output of `make docs`

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: docs/source/user_guide/pipeline-components.md [hunk 1]
```diff
@@ -80,23 +103,24 @@ To access the panel in JupyterLab:
...
+#### Adding a component catalog
 
-To add a component registry entry: 
+To add components from a catalog:
 
 1. Click `+` in the _Pipeline Components_ panel.
+1. Select a component catalog type from the list of available options.
 1. Define the registry entry. Refer to section [Configuration properties](#configuration-properties) for a description of each property.
 
 If the registry entry validates correctly, the associated pipeline components are added  to the pipeline editor's palette.
```
[the remark this answers is not in the document]
Comment (ptitzler) [reply docs/source/user_guide/pipeline-components.md:114]: s/registry/catalog

File: docs/source/user_guide/pipeline-components.md [hunk 2]
```diff
@@ -206,31 +230,53 @@ Example:
...
+ - `~patti/specs/filter_files/row_filter.yaml`
+
+Examples (CLI):
+ - `['/Users/patti/specs/load_data_from_public_source/http_operator.py']`
+ - `['~patti/specs/filter_files/row_filter.yaml']`
+ - `['/Users/patti/specs/comp1.yaml','/Users/patti/specs/comp2.yaml']`
+
+#### Directory component catalog
+
+The directory component catalog connector provides access to components that are stored in a file system directory:
```
[the remark this answers is not in the document]
Comment (ptitzler) [reply docs/source/user_guide/pipeline-components.md:256]: s/file system/filesystem/g

File: docs/source/user_guide/pipeline-components.md [hunk 3]
```diff
@@ -40,9 +40,32 @@ There are many example custom components available that you can utilize in pipel
...
+    Example: A URL component catalog that is configured using the `http://myserver:myport/mypath/my_component.yaml` URL makes the `my_component.yaml` component file available to Elyra.
+
+Refer to section [Built-in catalog connector reference](#built-in-catalog-connector-reference) for details about these connectors. 
+
+You can add support for other component catalogs by installing a connector from the [catalog connector marketplace](https://github.com/elyra-ai/examples/tree/master/component-catalog-connectors/connector-directory.md) or by [implementing your own catalog connector](https://github.com/elyra-ai/examples/tree/master/component-catalog-connectors/build-a-custom-connector.md).
+
 #### Example custom components
 
-For illustrative purposes the Elyra component registry includes a few custom components that you can use to get started. These example components and the generic components are pre-loaded into the pipeline editor palette by default.
+For illustrative purposes the Elyra includes a few custom components that you can use to get started. These example components and the generic components are pre-loaded into the pipeline editor palette by default.
```
Comment (kevin-bates) [inline docs/source/user_guide/pipeline-components.md:68]: reminder: this will need to be updated once the examples are moved out of the repo.

**Review by kevin-bates — COMMENTED** [review]
We might as well circle back on this once the dust has settled on the various PRs.

**Comment by ptitzler** [comment]
The documentation updates will be delivered in https://github.com/elyra-ai/elyra/pull/2286

**Pull request closed without merging**